pkg/syncer/admissionhandler/pvcsi_admissionhandler.go

```diff
@@ -93,6 +93,10 @@ func (h *CSIGuestWebhook) Handle(ctx context.Context, req admission.Request) (re
 			admissionResp := validatePVC(ctx, &req.AdmissionRequest)
 			resp.AdmissionResponse = *admissionResp.DeepCopy()
 		}
+	} else if req.Kind.Kind == "VolumeSnapshotClass" || req.Kind.Kind == "VolumeSnapshot" ||
+		req.Kind.Kind == "VolumeSnapshotContent" {
+		admissionResp := validateSnapshotOperationGuestRequest(ctx, &req.AdmissionRequest)
+		resp.AdmissionResponse = *admissionResp.DeepCopy()
```

Comment:
why do we don't have check for `if featureGateBlockVolumeSnapshotEnabled {` here just like we have it for `PersistentVolumeClaim` resource type.

Reply:
Talked to @deepakkinni 
We want to support other drivers in the guest cluster.